docs/design/2023-04-11-pause-user-ddl-when-upgrading.md

```diff
@@ -0,0 +1,109 @@
+# Proposal: Pause User DDL when Upgrading TiDB
+
+* Author: [zimulala](https://github.com/zimulala)
+* Tracking issue: https://github.com/pingcap/tidb/issues/39751
+
+## Abstract
+
+This document describes a feature that allows users to pause the execution of their Data Definition Language (DDL) statements during the upgrade process of TiDB. During the upgrade process, the system stops the execution of user DDL statements, completes the system DDL operation, and finally resumes the execution of the original user DDL statements. This feature is available for upgrades from v7.1 and later versions of TiDB, and it does not support upgrades from versions prior to v7.1.
+
+## Motivation or Background
+
+Since the cluster upgrade may need to deal with DDL statements(there may be some DDL statements being processed before the upgrade), and DDL itself implementation framework will also need to adjust, resulting in some versions may not be able to roll the upgrade situation. Although these cases only exist in some versions, it is not easy to describe them one by one. Therefore, when upgrading TiDB clusters, users need to confirm that there are no DDL statements being executed, otherwise there may be problems with undefined behavior.
+Now there are two problems:
+- The above rule is only described in the doc, and users may misoperate. Related [doc](https://docs.pingcap.com/tidb/dev/upgrade-tidb-using-tiup#upgrade-tidb-using-tiup).
+- During the upgrade operation, users need to communicate and coordinate with each business to cancel DDL, then upgrade, and finally re-execute DDL. Specific scenarios such as large cluster scenarios shared across multiple systems, such as scenarios on TiDB cloud.
+
+## Detailed Design
+
+Here are some roles for TiDB in the cluster (assuming upgrading from v7.1 to v7.2 here):
+- New TiDB: The new version of TiDB that will be upgraded, such as v7.2.
+- Old TiDB: Other TiDBs in the cluster that have not been upgraded, such as v7.1.
+- Owner TiDB: The TiDB where the DDL owner in the cluster is located.
+
+Specific upgrade process:
+1. New TiDB: Check whether the old TiDB version is greater than or equal to v7.1, if not, then exit the whole process.
+2. Notify the cluster that all TiDB enters to pause executing user DDL operations. Notification method:
+   1. New TiDB: Set the PD path (for example: `/tidb/server/global_state` , use this path later) to the current TiDB version. In addition, in order to prevent multiple TiDB nodes from upgrading at the same time, it will check whether the value on the `/tidb/server/global_state` path is empty.
+   2. All TiDBs：
+      1. The newly started TiDB will first check the `/tidb/server/global_state` value and whether it needs to enter upgrading_mode mode.
+      2. The running TiDB will watch `/tidb/server/global_state` , and when it receives an upgrade status notification, it will enter upgrading_mode mode.
```

Comment:
```suggestion
      2. The running TiDB will watch `/tidb/server/global_state` , and when it receives an upgrade status notification, it will enter the upgrading mode.
```